hypershift-operator/metrics_test.go

```diff
@@ -137,3 +142,150 @@ func TestMetrics(t *testing.T) {
 		})
 	}
 }
+
+func TestGetOperatorImage(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name                 string
+		podName              string
+		hypershiftPod        *corev1.Pod
+		hypershiftDeployment *appsv1.Deployment
+		expectedImage        string
+		expectedImageID      string
+	}{
+		{
+			name:    "Get hypershift operator image details from pod",
+			podName: "operator-fakepod",
+			hypershiftPod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "hypershift",
+				},
+				Status: corev1.PodStatus{
+					Conditions: []corev1.PodCondition{
+						{
+							Type:               corev1.PodReady,
+							Status:             corev1.ConditionTrue,
+							LastTransitionTime: metav1.Time{Time: time.Time{}.Add(10 * time.Minute)},
+						},
+					},
+					ContainerStatuses: []corev1.ContainerStatus{
+						{
+							Name:    assets.HypershiftOperatorName,
+							Image:   "quay.io/hypershift/hypershift:latest",
+							ImageID: "quay.io/hypershift/hypershift@sha256:746dc6b979cd6d42d4e6f3214e20eb5a2e0a2b4d1c1345f6914e5492e69d9afe",
+							State: corev1.ContainerState{
+								Running: &corev1.ContainerStateRunning{
+									StartedAt: metav1.Time{Time: time.Time{}},
+								},
+							},
+						},
+					},
+				},
+			},
+			hypershiftDeployment: &appsv1.Deployment{},
+			expectedImage:        "quay.io/hypershift/hypershift:latest",
+			expectedImageID:      "quay.io/hypershift/hypershift@sha256:746dc6b979cd6d42d4e6f3214e20eb5a2e0a2b4d1c1345f6914e5492e69d9afe",
+		},
+		{
+			name:    "Get hypershift operator image details from deployment",
+			podName: "operator-pod",
+			hypershiftPod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "hypershift",
+				},
+				Status: corev1.PodStatus{},
+			},
+			hypershiftDeployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      assets.HypershiftOperatorName,
+					Namespace: "hypershift",
+				},
+				Spec: appsv1.DeploymentSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  assets.HypershiftOperatorName,
+									Image: "quay.io/hypershift/hypershift:latest",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedImage:   "quay.io/hypershift/hypershift:latest",
+			expectedImageID: "not found",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.hypershiftPod.Name = tc.podName
+			t.Setenv("MY_NAMESPACE", "hypershift")
+			t.Setenv("MY_NAME", "operator-fakepod")
+			g := NewWithT(t)
+			client := fake.NewClientBuilder().WithScheme(api.Scheme).WithObjects(tc.hypershiftPod, tc.hypershiftDeployment).Build()
+			image, imageId, err := getOperatorImage(client)
+			g.Expect(err).NotTo(HaveOccurred(), "failed to get hypershift operator image details")
+			g.Expect(image).To(BeIdenticalTo(tc.expectedImage))
+			g.Expect(imageId).To(BeIdenticalTo(tc.expectedImageID))
+		})
+	}
+}
+
```

Comment:
this test can be dropped if we do https://github.com/openshift/hypershift/pull/2443/files#r1175068309